crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -799,7 +799,7 @@ where
             if let Expr::StringLiteral(ast::ExprStringLiteral { value, .. }) = expr {
                 self.deferred.string_type_definitions.push((
                     expr.range(),
-                    value,
+                    value.as_str(),
```

Comment:
Same for all these changes. If we could keep the `Deref` implementation in addition to the explicit `as_str` method, then code that doesn't need the more relaxed lifetime could be left as is.